### PR TITLE
[3.11] Index progress visibility

### PR DESCRIPTION
**Pull request by kvahed**
### Scope & Purpose

Enterprise companion PR: https://github.com/arangodb/enterprise/pull/1441
Backport of index https://github.com/arangodb/arangodb/pull/18516 on _api/index*

Also includes forward port of the changes included in the 3.10 PR https://github.com/arangodb/arangodb/pull/20530

- [ ] :hankey: Bugfix
- [X] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [X] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [X] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: *(Please link PR)*
  - [ ] Backport for 3.10: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: tests/js/client/shell/shell-index.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+///
+/// @author Kaveh Vahedipour
+////////////////////////////////////////////////////////////////////////////////
+
+const jsunity = require("jsunity");
+const internal = require("internal");
+const arango = internal.arango;
+const errors = internal.errors;
+const { helper, versionHas } = require("@arangodb/test-helper");
+const platform = require('internal').platform;
```
Comment (jsteemann) [inline tests/js/client/shell/shell-index.js:33]: Not used and can be removed.

File: tests/js/client/shell/shell-index.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  'use strict';
+
+  return {
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test: indexes
+////////////////////////////////////////////////////////////////////////////////
+
+    testIndexProgress : function () {
+      let c = internal.db._create("c0l", {numberOfShards: 3, replicationFactor: 2});
```
Comment (jsteemann) [inline tests/js/client/shell/shell-index.js:49]: It looks like the changes to the test are missing in the forward-port.

Comment (kvahed) [reply tests/js/client/shell/shell-index.js:49]: rechecking.

Comment (kvahed) [reply tests/js/client/shell/shell-index.js:49]: that is weird. the diff between the tests in 3.10/3.11 is empty.

File: tests/js/client/shell/shell-index.js (hunk 1 above)
Comment (kvahed) [reply tests/js/client/shell/shell-index.js:33]: done. thx

File: tests/js/client/shell/shell-index.js (hunk 2 above)
Comment (kvahed) [reply tests/js/client/shell/shell-index.js]: fixed

**Review by jsteemann — APPROVED** [review]
LGTM

File: tests/js/client/shell/shell-index.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        if (++count > 50) {
+          break;
+        }
+      }
+            
+      internal.debugSetFailAt("fillIndex::unpause");
+      
+      count = 0;
+      // wait until index appears, at most 30 seconds
+      while (++count < 300) {
```
Comment (jsteemann) [inline tests/js/client/shell/shell-index.js:143]:
```suggestion
      while (++count < 3000) {
```

**Review by neunhoef — APPROVED** [review]
LGTM

**Review by neunhoef — APPROVED** [review]
LGTM

**Pull request merged**